modules/ding_seo/ding_seo.module

```diff
@@ -370,71 +371,55 @@ function ding_seo_xmlsitemap_link_info() {
  */
 function ding_seo_update_sitemap() {
   try {
-    ding_seo_check_sitemap_size();
+    // Check wether to trim any links.
+    $number_of_objects_deleted = ding_seo_check_sitemap_size();
+
     $last_check = variable_get('ding_seo_sitemap_last_check', 0);
-    $objects = ding_seo_get_objects($last_check);
+    // We save the Request time before we do any database operations. So we dont miss
+    // any objects added after we started.
+    $now = REQUEST_TIME;
+
+    // Get the objects added since last check.
+    $objects = db_select('ting_object_sitemap', 't')
+      ->condition('t.created', $last_check, '>')
+      ->fields('t', ['xmlsitemap_link_id', 'ting_object_id'])
+      ->execute()
+      ->fetchAllKeyed();
 
     if (count($objects) > 0) {
+      // Insert the new links into the sitemap table.
       ding_seo_update_sitemap_table($objects);
+
       variable_set('xmlsitemap_regenerate_needed', TRUE);
-      variable_set('ding_seo_sitemap_last_check', time());
+      variable_set('ding_seo_sitemap_last_check', $now);
     }
-    watchdog('ding_seo', 'Updated sitemap: %time', array('%time' => REQUEST_TIME), WATCHDOG_INFO);
+    $replacements = ['%deleted' => $number_of_objects_deleted, '%added' => count($objects)];
+    watchdog('ding_seo', 'Updated sitemap. %deleted objects deleted. %added objects added',$replacements , WATCHDOG_INFO);
   }
   catch (Exception $exception) {
     watchdog_exception('ding_seo', $exception);
   }
 }
 
-/**
- * Get the new links from last update.
- */
-function ding_seo_get_objects($last_check) {
-  $objects = [];
-
-  $query = db_select('ting_object_sitemap', 't')
-    ->condition('t.created', $last_check, '>')
-    ->fields('t', ['tid', 'ting_object_id'])
-    ->execute();
-
-  while ($value = $query->fetchAssoc()) {
-    $tid = $value['tid'];
-    $objects[$tid] = $value['ting_object_id'];
-  }
-
-  return $objects;
-}
-
 /**
  * Insert the new links into xmlsitemaptable.
  */
 function ding_seo_update_sitemap_table($objects) {
   $maxid = (int) db_query("SELECT max(id) FROM {xmlsitemap} WHERE type = 'ting_object'")->fetchField() ?: 0;
-  $chunks = array_chunk($objects, 1000, TRUE);
-  foreach ($chunks as $chunk) {
-    // We could use xmlsitemap_link_save but it takes only one link at a time.
-    // For performance reasons we update the database table our selves.
-    $query = db_insert('xmlsitemap')
-      ->fields(array('id', 'type', 'loc', 'language', 'priority'));
-    foreach ($chunk as $id => $object) {
-      // If our id isn't the largest somthings wrong.
-      if ($id > $maxid) {
-        $query->values(
-          [
-            'id' => $id,
-            'type' => 'ting_object',
-            'loc' => 'ting/object/' . $object,
-            'language' => 'und',
-            'priority' => 0.5,
-          ]
-        );
-      }
+  foreach ($objects as $id => $object) {
+    // If our id isn't the largest somthings wrong.
```

Comment:
What is wrong and how did we end up here?

Reply:
Also spelling

Reply:
Removed it. Dont need it when we use xmlsitemap_save_link